### PR TITLE
cleanup Go linter settings

### PR DESCRIPTION
**Pull request by ferhatelmas**
## Description
* Tweak lint config
  * drop unused/wrong tool version and set version explicitly in makefile
  * run staticcheck via golangci concurrently
  * merge staticcheck config into golangci config Install tool locally under .bin folder.
* Use go1.19 as in go.mod.
* Fix issues from the correct versioning.

## Motivation and Context
Wrong/outdated versioning even if `1.43.0` was mentioned/required.

```
Installing golangci-lint
golangci/golangci-lint info checking GitHub for latest tag
golangci/golangci-lint info found version: 1.51.2 for v1.51.2/linux/amd64
golangci/golangci-lint info installed /home/runner/go/bin/golangci-lint
```

## How to test this PR?
Existing CI will do the job. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: internal/lsync/lrwmutex_test.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func testSimpleWriteLock(t *testing.T, duration time.Duration) (locked bool) {
 
 		lrwm.Unlock()
 	} else {
-		// fmt.Println("Write lock failed due to timeout")
+		t.Error("Write lock failed due to timeout")
```
Comment (harshavardhana) [inline internal/lsync/lrwmutex_test.go]: this change is wrong

File: internal/lsync/lrwmutex_test.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func testDualWriteLock(t *testing.T, duration time.Duration) (locked bool) {
 
 		lrwm.Unlock()
 	} else {
-		// fmt.Println("2nd write lock failed due to timeout")
+		t.Error("2nd write lock failed due to timeout")
```
Comment (harshavardhana) [inline internal/lsync/lrwmutex_test.go]: this change is wrong

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: internal/lsync/lrwmutex_test.go (hunk 1 above)
Comment (ferhatelmas) [reply internal/lsync/lrwmutex_test.go]: yes, realized from the failed CI, pushed a fix. Let me know if makes sense.

File: Makefile [hunk 3]
```diff
@@ -8,6 +8,9 @@ GOOS := $(shell go env GOOS)
 VERSION ?= $(shell git describe --tags)
 TAG ?= "minio/minio:$(VERSION)"
 
+GOLANGCI_VERSION = v1.51.2
+GOLANGCI = .bin/golangci/$(GOLANGCI_VERSION)/golangci-lint
```
Comment (harshavardhana) [inline Makefile]:
```suggestion
GOLANGCI_DIR = .bin/golangci/$(GOLANGCI_VERSION)/
GOLANGCI = $(GOLANGCI_DIR)/golangci-lint
```

File: Makefile [hunk 4]
```diff
@@ -19,10 +22,9 @@ help: ## print this help
 
 getdeps: ## fetch necessary dependencies
 	@mkdir -p ${GOPATH}/bin
-	@echo "Installing golangci-lint" && curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin
+	@echo "Installing golangci-lint" && curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(dir $(GOLANGCI)) $(GOLANGCI_VERSION)
```
Comment (harshavardhana) [inline Makefile]:
```suggestion
	@echo "Installing golangci-lint" && curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOLANGCI_DIR) $(GOLANGCI_VERSION)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (ferhatelmas) [reply Makefile]: updated, PTAL.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :x: [more...](#b1b9b7dc-8ba3-4277-8801-2805e0b3be7b) |
<a name="b1b9b7dc-8ba3-4277-8801-2805e0b3be7b"></a>
### 16736-ed3de2e/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.4:31383
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp a7db20080247:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 17 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 17 seconds
(8/14) Running minio-go tests ... done in 58 seconds
(9/14) Running minio-java tests ... done in 37 seconds
(10/14) Running minio-js tests ... FAILED in 57 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-e920debb-f019-4d67-8458-671062f4953f, events:s3:ObjectRemoved:*",
  "duration": 13278,
  "status": "FAIL",
  "error": "Error: read ETIMEDOUT at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 2 minutes and 18 seconds
(11/14) Running s3cmd tests ... done in 17 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 13 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: .golangci.yml [hunk 5]
```diff
@@ -1,39 +1,41 @@
...
-service:
-  golangci-lint-version: 1.43.0 # use the fixed version to not introduce new linters unexpectedly
+    - should have a package comment
+    - error strings should not be capitalized or end with punctuation or a newline
+    # todo fix these when we get enough time.
+    - 'singleCaseSwitch: should rewrite switch statement to if statement'
+    - 'unlambda: replace'
+    - 'captLocal:'
+    - 'ifElseChain:'
+    - 'elseif:'
```
Comment (klauspost) [inline .golangci.yml:41]: Not really sure why these are excluded. Oh well, we don't have to fix that here, tough.

**Review by klauspost — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: .golangci.yml (hunk 5 above)
Comment (ferhatelmas) [reply .golangci.yml:41]: It seems they will cause a bigger diff. Let me get rid of them separately and refactor relevant places.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#c951853b-2553-46c8-b773-b7b98893dd8f) |
<a name="c951853b-2553-46c8-b773-b7b98893dd8f"></a>
### 16736-92297f6/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:31600
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 5bc72cdbd3ec:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 3 minutes and 18 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 16 seconds
(8/14) Running minio-go tests ... done in 48 seconds
(9/14) Running minio-java tests ... done in 30 seconds
(10/14) Running minio-js tests ... done in 50 seconds
(11/14) Running minio-py tests ... FAILED in 1 minutes and 18 seconds
{
  "name": "minio-py:test_stat_object",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-5dedaf2a-3873-48e6-8fa2-caf3072699f6",
    "object_name": "c1321fcf-6511-408b-bb3b-87b55b2cede3-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "X-Amz-Meta-Testing": "value"
    },
    "content_type": "application/octet-stream"
  },
  "message": "('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 449, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 3, in raise_from\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 444, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.8/http/client.py\", line 1348, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.8/http/client.py\", line 316, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.8/http/client.py\", line 277, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"/usr/lib/python3.8/socket.py\", line 669, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 820, in test_stat_object\n    _test_stat_object(log_entry, sse)\n  File \"/mint/run/core/minio-py/tests.py\", line 794, in _test_stat_object\n    result = _CLIENT.put_object(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1766, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1725, in put_object\n    upload_id = self._create_multipart_upload(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1565, in _create_multipart_upload\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 550, in increment\n    raise six.reraise(type(error), error, _stacktrace)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/packages/six.py\", line 769, in reraise\n    raise value.with_traceback(tb)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 449, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 3, in raise_from\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 444, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.8/http/client.py\", line 1348, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.8/http/client.py\", line 316, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.8/http/client.py\", line 277, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"/usr/lib/python3.8/socket.py\", line 669, in readinto\n    return self._sock.recv_into(b)\nurllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))\n",
  "duration": 61767
}
(11/14) Running s3cmd tests ... done in 17 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 18 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**